plugin/pkg/admission/persistentvolume/resize/admission.go

```diff
@@ -98,21 +98,31 @@ func (pvcr *persistentVolumeClaimResize) Admit(a admission.Attributes) error {
 		return nil
 	}
 
-	// Growing Persistent volumes is only allowed for PVCs for which their StorageClass
-	// explicitly allows it
-	if !pvcr.allowResize(pvc, oldPvc) {
-		return admission.NewForbidden(a, fmt.Errorf("only dynamically provisioned pvc can be resized and "+
-			"the storageclass that provisions the pvc must support resize"))
-	}
-
-	// volume plugin must support resize
-	pv, err := pvcr.pvLister.Get(pvc.Spec.VolumeName)
-	if err != nil {
+	if pvc.Status.Phase != api.ClaimBound {
```

Comment:
this allows resizing before the claim is bound regardless of whether the storage class supports it?

Reply:
Not really, resizing of all unbound claims are prohibited by validation - https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L1599

Also, the code here checks if underlying volume type supports resizing and for unbound PVCs, we can't perform that check because they don't have a PV yet.

Reply:
this is looking at the status of the PVC object the user submitted, which is ignored and overridden with the status of the existing object.

Reply:
ack, I will change this to use `oldPvc`.

Reply:
can the new PVC specify both the volume name and increase the size?

Reply:
yes, that is possible. why?

Reply:
I have moved the check inside the if loop and returned forbidden error if user is trying to resize unbound claims.